mineral-ui/mineral-ui
# [website] Improve routing and some relative linking

### Description


- Fix new page (vs. same-page) anchor links via ScrollToIdOnMount
- Remove forced trailing slash from routes
- Add redirect for /components route
- Pass slug to ComponentDocExample and use it to generate relative-to-root links
- Fix broken anchor link in Component Theme table intro
- Convert same-page anchor links to routed ones when viewing standalone examples
- Change all relative links to be relative-to-the-root


### Motivation and context


Closes #533, connects #547 (/components route) and fixes a few other niggles.


### Screenshots, videos, or demo, if appropriate


https://533-fix-routed-anchors--mineral-ui.netlify.com/getting-started#installation

### How to test


1. Click the link above
1. Did you scroll to the Installation heading? Good!
1. Navigate to a component's theme variables table and click on either the "global" or "local" link.
1. Did you scroll to the appropriate heading? Good!
1. Navigate to a specific example route for a component that contains a same-page anchor link (like [this one](https://533-fix-routed-anchors--mineral-ui.netlify.com/components/box/width)).
1. Click the would-be anchor link and confirm you instead navigate to the example route
1. Click the back link and then click the same now-still-an-anchor link and confirm you scroll to the example
1. Confirm you can navigate to any given page with and without the trailing slash


### Types of changes

- Other (provide details below)

Just website stuff.


### Checklist




### How does this PR make you feel?

![60% of the time, it works everytime](https://media.giphy.com/media/LEKtRCGyA90QM/giphy.gif)
